src/pages/AppDetail.svelte

```diff
@@ -69,4 +100,15 @@
       </li>
     {/each}
   </ul>
+
+  {#if total > perPage}
+    <Pagination
+      {currentPage}
+      {from}
+      {lastPage}
+      {perPage}
+      {to}
+      {total}
+      on:changePage={(ev) => loadPage({ page: ev.detail })} />
+  {/if}
```

Comment:
I haven't thought about this too deeply, but this seems like a fairly complex interface. Might be worth looking at some other examples for a simpler implementation: https://github.com/fouita/svelte-tw-pagination

Reply:
@wlach the implementation is quite similar, however the interface here looks a little complex because we are working with `json` and not a paginated API so my idea is to get the JSON only once and use JavaScript to split it into the sections we need.

Reply:
In general it's a little better to do most post-processing (e.g. munging a json data structure) before sending any data to a component like this, as that keeps the interface simple (following the [seperation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns) design principle). Does that make sense?

Reply:
Managed to move core pagination functions to the pagination component

Reply:
This is better, however it's still a little complex. I think the pagination component really only needs two variables as input:

1. the total number of elements
2. the requested number of elements per page (optional, as mentioned above)
3. the current page that the user is on

The responsibility of rendering the current page of data can live in this component, maybe using a helper function defined in the pagination component that can take a full list of items and (2) and (3) above and return the list of items to be rendered.

Reply:
We have two extra params used to count items display on a page and show helpful text. "Showing x of n items" [It's not not completely necessary and be taken out]

Reply:
I think we're probably good here now, if we move to the `paginationState` approach that I mentioned above.